src/horizon/types/effects.ts

```diff
@@ -67,86 +67,86 @@ export enum EffectType {
   contract_credited = 96,
   contract_debited = 97
 }
-export interface BaseEffectRecord extends HorizonApi.BaseResponse {
+export interface BaseEffectRecord<T extends string = string> extends HorizonApi.BaseResponse {
   id: string;
   account: string;
   paging_token: string;
   type_i: EffectType;
-  type: string;
+  type: T;
   created_at: string;
 }
-export interface AccountCreated extends BaseEffectRecord {
+export interface AccountCreated extends BaseEffectRecord<'account_created'> {
   type_i: EffectType.account_created;
   starting_balance: string;
 }
-export interface AccountCredited extends BaseEffectRecord, OfferAsset {
+export interface AccountCredited extends BaseEffectRecord<'account_credited'>, OfferAsset {
   type_i: EffectType.account_credited;
   amount: string;
 }
-export interface AccountDebited extends BaseEffectRecord {
+export interface AccountDebited extends BaseEffectRecord<'account_debited'>, OfferAsset {
```

Comment:
Also took the opportunity to fix this, `AccountDebited` has `OfferAsset` fields.